fetpype/workflows/utils.py

```diff
@@ -180,15 +180,18 @@ def check_and_update_paths(data_dir, out_dir, nipype_dir, pipeline_name):
     if out_dir is None:
         out_dir = data_dir
 
+    if nipype_dir is None:
+        nipype_dir = out_dir
+
+    # derivatives
     out_dir = os.path.join(os.path.abspath(out_dir),
                            "derivatives", pipeline_name)
 
     os.makedirs(out_dir, exist_ok=True)
 
-    if nipype_dir is None:
-        nipype_dir = out_dir
-
-    nipype_dir = os.path.join(os.path.abspath(nipype_dir), pipeline_name)
+    # working directory
+    nipype_dir = os.path.join(os.path.abspath(nipype_dir),
+                              pipeline_name)
```

Comment:
I'd just change this to 
```
    nipype_dir = os.path.join(os.path.abspath(nipype_dir), "derivatives", "nipype",
                              pipeline_name)
```

to avoid writing the nipype_dir in the data_dir in case nothing is specified. We should not modify the user's data_dir I think.